### PR TITLE
Improve `SquaredL2SquaredAbsLoss`

### PR DESCRIPTION
**Pull request by bwohlberg**
Improve algorithm, docs, and testing of the cubic equation root calculation used by `SquaredL2SquaredAbsLoss`.

### CODE REVIEW

File: scico/test/functional/test_loss.py [hunk 1]
```diff
@@ -208,3 +208,12 @@ def test_prox(self, loss_tuple):
 
         pf = prox_test((1 + 1j) * snp.zeros(self.v.shape), L, L.prox, 0.0)  # complex zero v
         pf = prox_test((1 + 1j) * snp.zeros(self.v.shape), L, L.prox, 1.0)  # complex zero v
+
+
+def test_cubic_root():
+    N = 10000
+    p, _ = uniform(shape=(N,), dtype=snp.float32, minval=-10.0, maxval=10.0)
```
Comment (Michael-T-McCann) [inline scico/test/functional/test_loss.py:215]: consider seeding rng here so the test result is not stochastic

File: scico/test/functional/test_loss.py [hunk 2]
```diff
@@ -208,3 +208,12 @@ def test_prox(self, loss_tuple):
...
+    q, _ = uniform(shape=(N,), dtype=snp.float32, minval=-10.0, maxval=10.0)
```
Comment (Michael-T-McCann) [inline scico/test/functional/test_loss.py:216]: jax gotcha: I think p and q will be equal if you make them this way. Need to get the key from the first call and pass it to the second.

**Review by Michael-T-McCann — CHANGES_REQUESTED** [review]

File: scico/test/functional/test_loss.py (hunk 1 above)
Comment (bwohlberg) [reply scico/test/functional/test_loss.py]: Fixed.

File: scico/test/functional/test_loss.py (hunk 2 above)
Comment (bwohlberg) [reply scico/test/functional/test_loss.py]: Indeed they were. Fixed.

**Review by Michael-T-McCann — APPROVED** [review]

**Pull request merged**